src/Integration/LocalServices/SolutionWorkspaceService.cs

```diff
@@ -65,17 +65,18 @@ public IEnumerable<string> ListFiles()
         }
 
         [ExcludeFromCodeCoverage]
-        private IEnumerable<string> GetAllFilesInSolution(IVsSolution solution)
+        private ICollection<string> GetAllFilesInSolution(IVsSolution solution)
         {
-            IEnumerable<string> result = null;
+            ICollection<string> result = null;
             threadHandling.RunOnUIThread(() => result = GetLoadedProjects(solution)
                 .SelectMany(AllItemsInProject)
                 .Where(x => x != null)
                 .Where(x => x.Contains("\\"))
                 .Where(x => !x.EndsWith("\\"))
                 .Where(x => !x.Contains("\\.nuget\\"))
-                .Where(x => !x.Contains("\\node_modules\\"))); // move filtering closer to path extraction to avoid processing unnecessary items)
-
+                .Where(x => !x.Contains("\\node_modules\\"))
+                .ToHashSet(StringComparer.InvariantCultureIgnoreCase)); // move filtering closer to path extraction to avoid processing unnecessary items)
```

Comment:
I think an UT is missing for this deduplication logic.
It should show or explain by a comment the scenario under which duplicate files could be found.

Reply:
The UTs are missing for this whole class. I could refactor it to isolate the IVsHierarchy logic. Do you want me to do that in this PR or create a ticket?

Reply:
Ticket

Reply:
[#5405](https://github.com/SonarSource/sonarlint-visualstudio/issues/5405)
Opened in [SonarSource/sonarlint-visualstudio](https://github.com/SonarSource/sonarlint-visualstudio)